lib/hammer_cli_foreman/virtual_machine.rb

```diff
@@ -0,0 +1,57 @@
+
+module HammerCLIForeman
+
+  class VirtualMachine < HammerCLIForeman::Command
+    resource :compute_resources
+    command_name 'virtual_machine'
+    desc _("View and manage compute resource's virtual machines")
+
+    module ComputeResourceOptions
+
+      def self.included(base)
+        base.build_options
+
+        base.validate_options do
+          any(:option_compute_resource_id, :option_compute_resource_name).required
+        end
+      end
+
+    end
+
+    class PowerVmCommand < HammerCLIForeman::Command
+      action :power_vm
+      command_name 'power'
+
+      success_message _("Virtual machine is powering.")
+      failure_message _("Could not power the virtual machine")
+
+      build_options
+    end
+
+    class InfoCommand < HammerCLIForeman::InfoCommand
+      action :show_vm
+      command_name 'info'
+
+      output do
+        field :id, _("Id")
+        field :name, _("Name")
+      end
```

Comment:
There is a lot of information from the server for each virtual machine that we could get, but we show `Id` and `Name` only. Could we show all the information we're getting from the server?

Reply:
I actually checked it and it is a bit problematic because in each compute resource the data is named differently. For example, in ovirt the number of CPUs is under: 'cpu' while in vmware it is under: 'cpus'.
Do you maybe have an idea how to overcome this issue?

Reply:
Yes, we've got method called [provider_specific_fields](https://github.com/theforeman/hammer-cli-foreman/blob/d214945f09f88f4270476d162b93fac2c92ff66d/lib/hammer_cli_foreman/compute_resource/base.rb#L1) which allows you specify which fields for which CR should be displayed by [redefining it](https://github.com/theforeman/hammer-cli-foreman/blob/d214945f09f88f4270476d162b93fac2c92ff66d/lib/hammer_cli_foreman/compute_resource/gce.rb#L22).

But this assumes that we've got `'provider'` item in the response.

Reply:
Ok, sounds good, I will check if the response contains 'provider' item.

Reply:
There isn't a 'provider' item in the response. Is it necessary in order to use the relevant method?

Reply:
Not hard at all, one-row addition in the compute-resource API. I will try to add it.